packages/codemods/src/codemods/v2.2.x/updateClerk/updateClerk.ts

```diff
@@ -0,0 +1,61 @@
+import type { FileInfo, API } from 'jscodeshift'
+
+export default function transform(file: FileInfo, api: API) {
+  const j = api.jscodeshift
+
+  const ast = j(file.source)
+
+  ast
+    .find(j.ImportDeclaration, { source: { value: '@clerk/clerk-react' } })
+    .forEach((importDeclaration) => {
+      importDeclaration?.value.specifiers?.forEach((specifier) => {
+        if (
+          j.ImportSpecifier.check(specifier) &&
+          specifier.imported.name === 'withClerk'
+        ) {
+          // Found `withClerk` import. Now I want to replace that with a
+          // `ClerkLoaded` import instead
+          specifier.imported.name = 'ClerkLoaded'
+
+          // And finally, for the imports, we need to add a new import to give us `navigate`
+          j(importDeclaration).insertAfter(
+            "import { navigate } from '@redwoodjs/router'"
+          )
+        }
+      })
+    })
+
+  // Remove old RW Clerk components
+  ast.find(j.VariableDeclaration).forEach((variableDeclaration) => {
+    if (
+      variableDeclaration.value.declarations.find((declaration) => {
+        return (
+          j.VariableDeclarator.check(declaration) &&
+          j.Identifier.check(declaration.id) &&
+          (declaration.id.name === 'ClerkAuthProvider' ||
+            declaration.id.name === 'ClerkAuthConsumer')
+        )
+      })
+    ) {
+      j(variableDeclaration).remove()
+    }
+  })
+
+  ast.find(j.VariableDeclaration).insertBefore(`
```

Comment:
This `ast.find` needs to be a little more specific. We look for something like `const App = ...` specifically. There could be many other declarations besides the ClerkAuthConsumer or Provider.

Reply:
This is what I do now:

```
  const appVariableDeclaration = ast.find(j.VariableDeclaration, {
    declarations: [
      {
        id: { name: 'App' },
      },
    ],
  })
```